Cargo.toml

```diff
@@ -15,21 +15,36 @@ all-features = true
 [features]
 deflate = ["flate2"]
 
+# This feature enables the use of uninitialised memory for the deflate
+# extension. When compressing or decompressing, an uninitialised byte
+# slice will be used for writing the resulting bytes, assuming that
+# flate2 really only writes to and does not read from this memory.
+deflate_with_uninitialised_memory = []
+
+# This feature enables the use of uninitialised memory in conjunction with
+# `AsyncRead::read`. It relies on the implementation of `AsyncRead` to not
+# read from but only to write to this byte slice. We used to use
+# `AsyncRead::initialize` and the resulting `futures::io::Initializer`
+# for this purpose, but the later became an unstable API beginning
+# with futures-preview 0.3.0-alpha.19 so by default we now do zero the
+# byte slice before calling `read` unless this feature is used.
+read_with_uninitialised_memory = []
```

Comment:
I don't really understand what you're achieving with that. The `Initializer` thing allows a crate to say "trust me that you don't have to zero the buffer". Enabling this feature essentially means "all the crates you will ever use underneath soketto are trusted and don't need a zero'ed buffer, even through indirect dependencies".

Reply:
I am not sure when (if ever) the `read_intitializer` feature will be stabilised and I assume it will happen much later that the async/await release on stable. Therefore I thought it would be better to always zero memory and only provide the `read_with_unitialised_memory` feature as a way for clients to turn this off. Ultimately it may only be safe to do so in applications.